### ISSUE TITLE
Automatic case splitting fails when '-' is a constructor

### ISSUE THREAD

**Issue by astump** [agda/agda#3780]
If one has a datatype where '-' (no quotes) is a constructor, then automatic case splitting on an implicit argument of that type fails because it generates the pattern '{ -}' where the '-}' looks like end-of-quotation.  Here is a self-contained example, where doing Control-C Control-C in the hole in the body of myfun will generate that bad pattern '{ -}':

```
module split-minus where

data pol : Set where
  + : pol
  - : pol

myfun : {x : pol} → pol
myfun{x} = ?
```

**Comment by gallais** [comment]
Good news: it's fixed in 2.6.0 already! #3161 fixed the fix for #269 so that
constructors either starting or ending with `-` do not lead to parse errors
in this situation.

**Comment by astump** [comment]
Excellent!  Sorry I did not check the most recent version.
